packages/discord.js/src/sharding/Shard.js

```diff
@@ -406,6 +414,28 @@ class Shard extends EventEmitter {
 
     if (respawn) this.spawn(timeout).catch(err => this.emit('error', err));
   }
+
+  /**
+   * Increments max listeners by one for a given emitter, if they are not zero.
+   * @private
+   */
+  incrementMaxListeners(emitter) {
+    const maxListeners = emitter.getMaxListeners();
+    if (maxListeners !== 0) {
```

Comment:
```suggestion
    if (maxListeners) {
```
can't we do this?

Reply:
The value can be set to Infinity (or 0) to indicate an unlimited number of listeners.
source: https://nodejs.org/api/events.html#emittersetmaxlistenersn
So It's correct to check for that.

Reply:
Yeah I just saw it.